Final structural adjustments and schema cleanup

thebingi opened the pull request:
Final structural adjustments and schema cleanup

thebingi committed:
Final structural adjustments and schema cleanup

thebingi committed:
Required added to properties in dayAmountItem

thebingi commented:
Eftir að hafa rennt betur yfir staðalinn í lokin þá tók ég eftir nokkrum atriðum sem mætti bæta. PR er tilbúið og inniheldur eftirfarandi breytingar:

Uppfærður texti í lýsingu efst. Nánari útskýring á því hvernig þetta á að virka, þar sem ekki er til sérstakur endpoint fyrir milliinnheimtuaðila.
 
PayorID bætt við sem parameter til að sækja greiðslur.
Þetta er þegar í boði í kröfupottinum og gæti nýst einhverjum.
 
Paging bætt við aðgerðina þar sem sóttar eru upplýsingar um bunka.
Bunkar geta oft verið stórir og því mikilvægt að bjóða upp á paging þar sem staða á hverri kröfu er skilað.
 
ReferenceDate sett sem required hjá discount, defaultInterest og defaultCharge.
Þetta er required í kröfupottinum.
 
Svæðin í dayAmountItem sett sem required: chargeType, day og value.
Þetta er einnig required í kröfupottinum.
 
Bættum print nýlega við svarið þegar krafa er stofnuð. Því er eðlilegt að þetta sé einnig til staðar þegar upplýsingar um kröfu/kröfur eru sóttar.
 
CollectionStatus - WaitingForSecondary fjarlægt.
Þetta virðist hafa verið hugsað fyrir milliinnheimtuaðila til að sjá hvaða kröfur séu að fara í milliinnheimtu. Kröfupotturinn býður ekki upp á þessa yfirsýn. Það er lokað á milli kröfuhafa og milliinnheimtuaðila.  Eiga milliinnheimtuaðilar að geta séð þessar upplýsingar. Finnst þetta á gráu svæði.

halldorvagn commented:
Síðasta, CollectionStatus, það þarf að staðfesta frá öllum bönkum að þessi hegðun sé ekki til staðar.
Man þegar þetta var búið til á sínum tíma...

thebingi commented:
> Síðasta, CollectionStatus, það þarf að staðfesta frá öllum bönkum að þessi hegðun sé ekki til staðar. Man þegar þetta var búið til á sínum tíma...

Þetta er amk ekki í B2B í dag. Hvorki 2005 eða 2013.

asgeirhall commented:
Án þess að vita það fyrir víst þá held ég að WaitingForSecondary sé vegna Landsbankans.  Þeir eru einir um að láta milliinnheimtuaðila samþykkja kröfur í milliinnheimtu.

thebingi committed:
Fields added that can be used for filtering in GetClaims:
PaymentDate: This already exists in the claims pool, so it’s straightforward to add it as a filter option.
TemplateTransferDate: The date when a claim was moved to another template. This should also be relatively simple to add as a filter field. This is necessary for collection agencies to search by date, as it is currently more complicated since there is only a single endpoint.

thebingi commented:
Svæðum bætt við sem hægt er að nota til að filtera GetClaims:

**PaymentDate**
Þetta er þegar til staðar í kröfupottinum, þannig að það er einfalt að bæta því við sem filter valmöguleika.

**TemplateTransferDate**
Dagsetning þegar krafa var færð yfir á annað template. Þetta ætti einnig að vera tiltölulega einfalt að bæta við. Þetta er þægilegt fyrir milliinnheimtuaðila sem vilja leita eftir dagsetningu, þar sem enginn sérstakur endapunktur er til fyrir þá. Eins og áður.

thebingi commented:
Ég sótti eintak af TS 162:2013 en sé ekki „WaitingForSecondary“. Er þetta kannski úr eldri útgáfu staðalsins?
Ef þetta er sérlausn hjá Landsbankanum, þá velti ég fyrir mér hvort þetta eigi heima í staðlinum. Þetta gæti frekar verið sérútfærsla hjá Landsbankanum, sérstaklega þar sem aðrir virðast ekki vera að nota þetta.

Steinarlogis commented:
Eftir yfirferð hér í Landsbankanum hefur verið fallist á það að fjarlægja CollectionStatus "WaitingForSecondary" út úr staðlinum

kristinnstefansson commented:
Meeting TN-FMÞ-VH7-39 accepted this PR though there are adjustments that potentially will be accepted due to changes on RB functionality.
